src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -121,6 +121,48 @@ void cfApiSendTransferInfo(const CF_CONNECTION_KEY &connectionKey, const CF_TRAN
     }
 }
 
+void cfApiSendPlaceholdersTransferInfo(const CF_CONNECTION_KEY &connectionKey, const CF_TRANSFER_KEY &transferKey, NTSTATUS status, qint64 currentPlaceholdersCount, qint64 totalPlaceholdersCount)
+{
+    CF_OPERATION_INFO opInfo = { 0 };
+    CF_OPERATION_PARAMETERS opParams = { 0 };
+
+    opInfo.StructSize = sizeof(opInfo);
+    opInfo.Type = CF_OPERATION_TYPE_TRANSFER_PLACEHOLDERS;
+    opInfo.ConnectionKey = connectionKey;
+    opInfo.TransferKey = transferKey;
+
+    opParams.ParamSize = CF_SIZE_OF_OP_PARAM(TransferPlaceholders);
+    opParams.TransferPlaceholders.Flags = CF_OPERATION_TRANSFER_PLACEHOLDERS_FLAG_DISABLE_ON_DEMAND_POPULATION;
+    opParams.TransferPlaceholders.CompletionStatus = status;
+    opParams.TransferPlaceholders.PlaceholderTotalCount.QuadPart = totalPlaceholdersCount;
+    opParams.TransferPlaceholders.PlaceholderCount = currentPlaceholdersCount;
+    opParams.TransferPlaceholders.EntriesProcessed = currentPlaceholdersCount;
+    opParams.TransferPlaceholders.PlaceholderArray = nullptr;
+
+    const qint64 cfExecuteresult = CfExecute(&opInfo, &opParams);
+    if (cfExecuteresult != S_OK) {
+        qCCritical(lcCfApiWrapper) << "Couldn't send transfer info" << QString::number(transferKey.QuadPart, 16) << ":" << cfExecuteresult << QString::fromWCharArray(_com_error(cfExecuteresult).ErrorMessage());
+    }
+
+    const auto isDownloadFinished = ((offset + currentBlockLength) == totalLength);
+    if (isDownloadFinished) {
+        return;
+    }
+
+           // refresh Windows Copy Dialog progress
+    LARGE_INTEGER progressTotal;
+    progressTotal.QuadPart = totalLength;
+
+    LARGE_INTEGER progressCompleted;
+    progressCompleted.QuadPart = offset;
+
+    const qint64 cfReportProgressresult =  CfReportProviderProgress(connectionKey, transferKey, progressTotal, progressCompleted);
+
+    if (cfReportProgressresult != S_OK) {
+        qCCritical(lcCfApiWrapper) << "Couldn't report provider progress" << QString::number(transferKey.QuadPart, 16) << ":" << cfReportProgressresult << QString::fromWCharArray(_com_error(cfReportProgressresult).ErrorMessage());
+    }
```

Comment:
guess this block is not required for the `TransferPlaceholders` operation

Reply:
yes
will double check

Reply:
done